mlrun/api/db/sqldb/db.py

```diff
@@ -363,48 +405,47 @@ def _store_artifact(
     ):
         project = project or config.default_project
         artifact = deepcopy(artifact)
-        updated = artifact.get("updated")
-        if not updated:
-            updated = artifact["updated"] = datetime.now(timezone.utc)
-        db_key = artifact.get("db_key")
-        if db_key and db_key != key:
-            raise mlrun.errors.MLRunInvalidArgumentError(
-                "Conflict between requested key and key in artifact body"
+        if "metadata" in artifact:
+            updated, key, labels = self._process_stored_artifact_dict(
+                artifact, key, iter
             )
-        if not db_key:
-            artifact["db_key"] = key
-        if iter:
-            key = f"{iter}-{key}"
+        else:
+            updated, key, labels = self._process_stored_legacy_artifact_dict(
+                artifact, key, iter
+            )
+
         art = self._get_artifact(session, uid, project, key)
-        labels = artifact.get("labels", {})
         if not art:
             art = Artifact(key=key, uid=uid, updated=updated, project=project)
         update_labels(art, labels)
 
-        # Ensure there is no "tag" field in the object, to avoid inconsistent situations between
-        # body and tag parameter provided.
-        artifact.pop("tag", None)
-
         art.struct = artifact
         self._upsert(session, art)
         if tag_artifact:
             tag = tag or "latest"
             self.tag_artifacts(session, [art], project, tag)
 
+    @staticmethod
+    def _set_tag_in_artifact_struct(artifact, tag):
+        if "metadata" in artifact:
```

Comment:
This heuristic is repeating a lot, export to a method `_is_legacy_format_artifact_dict()`